app/models/signon_identity.rb

```diff
@@ -79,13 +88,26 @@ def logout_url(redirect_to: nil)
     url.to_s
   end
 
+  def accessible_estates
+    @_ae ||= self.class.accessible_estates(@permissions).order(:nomis_id).to_a
+  end
+
+  def accessible_estate?(estate)
+    accessible_estates.include?(estate)
+  end
+
+  def default_estate
+    accessible_estates.first || fail('Should never be nil')
```

Comment:
It won't be nil in that case because the `self.class.accessible_estates` method is called on login and validated not to be empty. So in that case the user won't be logged in.

There is an edge case if an estate is removed from PVB after the user has logged in, but removing estates is not something we've ever done. I raised an exception so that the issue would be clear if it ever happened, rather than passing nils elsewhere. It could be handled properly but I'm not sure it's worthwhile.